plugin/mutate/README.md

```diff
@@ -1,78 +1,105 @@
-```
+The Dextool mutation testing plugin. 
 
-The different states a mutant are found in Mutation.Kind.
+## Overview
 
-Reset all mutations of a kind to unknown which forces them to be tested again:
-```sh
-sqlite3 dextool_mutate.sqlite3 "update mutation SET status=0 WHERE mutation.kind=FOO"
-```
+Dextool's plugin for mutation testing of C/C++ projects. It can help you design new tests and evaluate the quality of existing  tests.
+
+### Features
+
+TODO: List supported mutation operators etc.
+TODO: What are the selling points etc?
 
-# Mutation Testing of Google Test
+# Getting Started
 
-This is an example of how to mutation test google test itself.
-It assumes the current directory is _build_ which is then located the google test repo.
+If you are new to dextool, we suggest you to read the following:
 
-Create a database of all mutation points:
+* TODO: Something about compilation databases
+* TODO: Something about mutation testing
+
+## Mutation testing
+
+### Using CMake
+
+The [Google Test project](https://github.com/google/googletest) will be used as an example.
+
+When setting up Dextool Mutate to analyze a CMake project, the typical workflow is to:
 ```sh
-cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
-dextool mutate analyze --compile-db compile_commands.json --restrict .. -- -D_POSIX_PATH_MAX=1024
+# Obtain the project
+git clone https://github.com/google/googletest.git
+cd googletest
+
+# Create a directory to hold the build output:
+mkdir build
+cd build
+
+# Generate a JSON compilation database and build scripts:
+cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -Dgtest_build_tests=ON -Dgmock_build_tests=ON ..
+make
 ```
 
-Reconfigure and prebuild with the tests activated:
+Suppose the source code is in ... and the test code is elsewhere. Use the ``--restrict`` option to specify what to analyze:
 ```sh
-cmake -Dgtest_build_tests=ON -Dgmock_build_tests=ON ..
-make
+dextool mutate analyze --compile-db compile_commands.json --restrict .. -- -D_POSIX_PATH_MAX=1024
 ```
 
-Create a file tester.sh with this content:
+Create a script ``tester.sh`` that runs the entire test suite when invoked:
 ```sh
 #!/bin/bash
 set -e
 make test ARGS="-j$(nproc)"
 ```
 
-Create a file compile.sh with this content:
+Create a script ``compile.sh`` that builds the project when invoked:
 ```sh
 #!/bin/bash
 set -e
 make -j$(nproc)
 ```
 
-Make them executable so they can be used by dextool:
+Make the scripts executable so they can be run by dextool:
 ```sh
 chmod 755 tester.sh
 chmod 755 compile.sh
 ```
 
-Start mutation testing!!!!:
+Execute the mutation testing:
 ```sh
 dextool mutate test --mutant-test ./tester.sh --mutant-compile ./compile.sh --restrict ..
 ```
 
-It is possible to run multiple `test` against the same database.
-Just make sure they don't mutate the same source code.
-
-To see the result:
+Generate the mutation testing result:
 ```sh
 dextool mutate report --restrict .. --level alive --mutant lcr
 ```
 
-## Compiling Google Test with Coverage
```

Comment:
Maybe keep the headline to "separate" this chapter from the "mutation testing"?
It is technically _not_ part of the mutation testing but something that is probably interesting for the reader to look at.